<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">

<html lang="en">

<head>
  <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
  <title>LCOV - lcov.info - utils\api\lib\src\data\custom_exception.dart</title>
  <link rel="stylesheet" type="text/css" href="../../../../../gcov.css">
</head>

<body>

  <table width="100%" border=0 cellspacing=0 cellpadding=0>
    <tr><td class="title">LCOV - code coverage report</td></tr>
    <tr><td class="ruler"><img src="../../../../../glass.png" width=3 height=3 alt=""></td></tr>

    <tr>
      <td width="100%">
        <table cellpadding=1 border=0 width="100%">
          <tr>
            <td width="10%" class="headerItem">Current view:</td>
            <td width="35%" class="headerValue"><a href="../../../../../index.html">top level</a> - <a href="index.html">utils\api\lib\src\data\custom_exception.dart</a> - custom_exception.dart<span style="font-size: 80%;"> (source / <a href="custom_exception.dart.func-sort-c.html">functions</a>)</span></td>
            <td width="5%"></td>
            <td width="15%"></td>
            <td width="10%" class="headerCovTableHead">Hit</td>
            <td width="10%" class="headerCovTableHead">Total</td>
            <td width="15%" class="headerCovTableHead">Coverage</td>
          </tr>
          <tr>
            <td class="headerItem">Test:</td>
            <td class="headerValue">lcov.info</td>
            <td></td>
            <td class="headerItem">Lines:</td>
            <td class="headerCovTableEntry">0</td>
            <td class="headerCovTableEntry">66</td>
            <td class="headerCovTableEntryLo">0.0 %</td>
          </tr>
          <tr>
            <td class="headerItem">Date:</td>
            <td class="headerValue">Thu Dec  8 17:15:41 2022</td>
            <td></td>
            <td class="headerItem">Functions:</td>
            <td class="headerCovTableEntry">0</td>
            <td class="headerCovTableEntry">0</td>
            <td class="headerCovTableEntryHi">-</td>
          </tr>
          <tr><td><img src="../../../../../glass.png" width=3 height=3 alt=""></td></tr>
        </table>
      </td>
    </tr>

    <tr><td class="ruler"><img src="../../../../../glass.png" width=3 height=3 alt=""></td></tr>
  </table>

  <table cellpadding=0 cellspacing=0 border=0>
    <tr>
      <td><br></td>
    </tr>
    <tr>
      <td>
<pre class="sourceHeading">          Line data    Source code</pre>
<pre class="source">
<a name="1"><span class="lineNum">       1 </span>            : // ignore_for_file: public_member_api_docs, sort_constructors_first</a>
<a name="2"><span class="lineNum">       2 </span>            : // ignore_for_file: constant_identifier_names, library_private_types_in_public_api</a>
<a name="3"><span class="lineNum">       3 </span>            : import 'package:common_dependency/common_dependency.dart';</a>
<a name="4"><span class="lineNum">       4 </span>            : import 'package:flutter/foundation.dart';</a>
<a name="5"><span class="lineNum">       5 </span>            : </a>
<a name="6"><span class="lineNum">       6 </span>            : /// An enum that holds names for our custom exceptions.</a>
<a name="7"><span class="lineNum">       7 </span>            : enum _ExceptionType {</a>
<a name="8"><span class="lineNum">       8 </span>            :   /// The exception for an expired bearer token.</a>
<a name="9"><span class="lineNum">       9 </span>            :   TokenExpiredException,</a>
<a name="10"><span class="lineNum">      10 </span>            :   Unauthorized,</a>
<a name="11"><span class="lineNum">      11 </span>            : </a>
<a name="12"><span class="lineNum">      12 </span>            :   /// The exception for a prematurely cancelled request.</a>
<a name="13"><span class="lineNum">      13 </span>            :   CancelException,</a>
<a name="14"><span class="lineNum">      14 </span>            : </a>
<a name="15"><span class="lineNum">      15 </span>            :   /// The exception for a failed connection attempt.</a>
<a name="16"><span class="lineNum">      16 </span>            :   ConnectTimeoutException,</a>
<a name="17"><span class="lineNum">      17 </span>            : </a>
<a name="18"><span class="lineNum">      18 </span>            :   /// The exception for failing to send a request.</a>
<a name="19"><span class="lineNum">      19 </span>            :   SendTimeoutException,</a>
<a name="20"><span class="lineNum">      20 </span>            : </a>
<a name="21"><span class="lineNum">      21 </span>            :   /// The exception for failing to receive a response.</a>
<a name="22"><span class="lineNum">      22 </span>            :   ReceiveTimeoutException,</a>
<a name="23"><span class="lineNum">      23 </span>            : </a>
<a name="24"><span class="lineNum">      24 </span>            :   /// The exception for no internet connectivity.</a>
<a name="25"><span class="lineNum">      25 </span>            :   SocketException,</a>
<a name="26"><span class="lineNum">      26 </span>            : </a>
<a name="27"><span class="lineNum">      27 </span>            :   /// A better name for the socket exception.</a>
<a name="28"><span class="lineNum">      28 </span>            :   FetchDataException,</a>
<a name="29"><span class="lineNum">      29 </span>            : </a>
<a name="30"><span class="lineNum">      30 </span>            :   /// The exception for an incorrect parameter in a request/response.</a>
<a name="31"><span class="lineNum">      31 </span>            :   FormatException,</a>
<a name="32"><span class="lineNum">      32 </span>            : </a>
<a name="33"><span class="lineNum">      33 </span>            :   /// The exception for an unknown type of failure.</a>
<a name="34"><span class="lineNum">      34 </span>            :   UnrecognizedException,</a>
<a name="35"><span class="lineNum">      35 </span>            : </a>
<a name="36"><span class="lineNum">      36 </span>            :   /// The exception for an unknown exception from the api.</a>
<a name="37"><span class="lineNum">      37 </span>            :   ApiException,</a>
<a name="38"><span class="lineNum">      38 </span>            : </a>
<a name="39"><span class="lineNum">      39 </span>            :   /// The exception for any parsing failure encountered during</a>
<a name="40"><span class="lineNum">      40 </span>            :   /// serialization/deserialization of a request.</a>
<a name="41"><span class="lineNum">      41 </span>            :   SerializationException,</a>
<a name="42"><span class="lineNum">      42 </span>            : }</a>
<a name="43"><span class="lineNum">      43 </span>            : </a>
<a name="44"><span class="lineNum">      44 </span>            : class CustomException extends Equatable implements Exception {</a>
<a name="45"><span class="lineNum">      45 </span>            :   final String name, message;</a>
<a name="46"><span class="lineNum">      46 </span>            :   final String? code;</a>
<a name="47"><span class="lineNum">      47 </span>            :   final int? statusCode;</a>
<a name="48"><span class="lineNum">      48 </span>            :   final Failure failure;</a>
<a name="49"><span class="lineNum">      49 </span>            :   final _ExceptionType exceptionType;</a>
<a name="50"><span class="lineNum">      50 </span>            : </a>
<a name="51"><span class="lineNum">      51 </span><span class="lineNoCov">          0 :   CustomException({</span></a>
<a name="52"><span class="lineNum">      52 </span>            :     this.code,</a>
<a name="53"><span class="lineNum">      53 </span>            :     int? statusCode,</a>
<a name="54"><span class="lineNum">      54 </span>            :     required this.message,</a>
<a name="55"><span class="lineNum">      55 </span>            :     Failure? failureType,</a>
<a name="56"><span class="lineNum">      56 </span>            :     this.exceptionType = _ExceptionType.ApiException,</a>
<a name="57"><span class="lineNum">      57 </span>            :   })  : statusCode = statusCode ?? 500,</a>
<a name="58"><span class="lineNum">      58 </span><span class="lineNoCov">          0 :         name = exceptionType.name,</span></a>
<a name="59"><span class="lineNum">      59 </span><span class="lineNoCov">          0 :         failure = failureType ?? UnRecognizedFailure();</span></a>
<a name="60"><span class="lineNum">      60 </span>            : </a>
<a name="61"><span class="lineNum">      61 </span><span class="lineNoCov">          0 :   factory CustomException.fromDioException(Exception error) {</span></a>
<a name="62"><span class="lineNum">      62 </span>            :     try {</a>
<a name="63"><span class="lineNum">      63 </span><span class="lineNoCov">          0 :       if (error is DioError) {</span></a>
<a name="64"><span class="lineNum">      64 </span><span class="lineNoCov">          0 :         switch (error.type) {</span></a>
<a name="65"><span class="lineNum">      65 </span><span class="lineNoCov">          0 :           case DioErrorType.cancel:</span></a>
<a name="66"><span class="lineNum">      66 </span><span class="lineNoCov">          0 :             return CustomException(</span></a>
<a name="67"><span class="lineNum">      67 </span>            :               exceptionType: _ExceptionType.CancelException,</a>
<a name="68"><span class="lineNum">      68 </span><span class="lineNoCov">          0 :               statusCode: error.response?.statusCode,</span></a>
<a name="69"><span class="lineNum">      69 </span>            :               message: 'Request cancelled prematurely',</a>
<a name="70"><span class="lineNum">      70 </span>            :             );</a>
<a name="71"><span class="lineNum">      71 </span><span class="lineNoCov">          0 :           case DioErrorType.connectTimeout:</span></a>
<a name="72"><span class="lineNum">      72 </span><span class="lineNoCov">          0 :             return CustomException(</span></a>
<a name="73"><span class="lineNum">      73 </span>            :               exceptionType: _ExceptionType.ConnectTimeoutException,</a>
<a name="74"><span class="lineNum">      74 </span><span class="lineNoCov">          0 :               statusCode: error.response?.statusCode,</span></a>
<a name="75"><span class="lineNum">      75 </span>            :               message: 'Connection not established',</a>
<a name="76"><span class="lineNum">      76 </span><span class="lineNoCov">          0 :               failureType: NetworkFailure(),</span></a>
<a name="77"><span class="lineNum">      77 </span>            :             );</a>
<a name="78"><span class="lineNum">      78 </span><span class="lineNoCov">          0 :           case DioErrorType.sendTimeout:</span></a>
<a name="79"><span class="lineNum">      79 </span><span class="lineNoCov">          0 :             return CustomException(</span></a>
<a name="80"><span class="lineNum">      80 </span>            :               exceptionType: _ExceptionType.SendTimeoutException,</a>
<a name="81"><span class="lineNum">      81 </span><span class="lineNoCov">          0 :               statusCode: error.response?.statusCode,</span></a>
<a name="82"><span class="lineNum">      82 </span>            :               message: 'Failed to send',</a>
<a name="83"><span class="lineNum">      83 </span><span class="lineNoCov">          0 :               failureType: NetworkFailure(),</span></a>
<a name="84"><span class="lineNum">      84 </span>            :             );</a>
<a name="85"><span class="lineNum">      85 </span><span class="lineNoCov">          0 :           case DioErrorType.receiveTimeout:</span></a>
<a name="86"><span class="lineNum">      86 </span><span class="lineNoCov">          0 :             return CustomException(</span></a>
<a name="87"><span class="lineNum">      87 </span>            :               exceptionType: _ExceptionType.ReceiveTimeoutException,</a>
<a name="88"><span class="lineNum">      88 </span><span class="lineNoCov">          0 :               statusCode: error.response?.statusCode,</span></a>
<a name="89"><span class="lineNum">      89 </span>            :               message: 'Failed to receive',</a>
<a name="90"><span class="lineNum">      90 </span><span class="lineNoCov">          0 :               failureType: NetworkFailure(),</span></a>
<a name="91"><span class="lineNum">      91 </span>            :             );</a>
<a name="92"><span class="lineNum">      92 </span><span class="lineNoCov">          0 :           case DioErrorType.response:</span></a>
<a name="93"><span class="lineNum">      93 </span><span class="lineNoCov">          0 :             if (error.response?.statusCode == 401) {</span></a>
<a name="94"><span class="lineNum">      94 </span><span class="lineNoCov">          0 :               return CustomException(</span></a>
<a name="95"><span class="lineNum">      95 </span>            :                 exceptionType: _ExceptionType.Unauthorized,</a>
<a name="96"><span class="lineNum">      96 </span><span class="lineNoCov">          0 :                 statusCode: error.response?.statusCode,</span></a>
<a name="97"><span class="lineNum">      97 </span>            :                 message: 'Unauthorized user',</a>
<a name="98"><span class="lineNum">      98 </span><span class="lineNoCov">          0 :                 failureType: UnAuthorizedFailure(),</span></a>
<a name="99"><span class="lineNum">      99 </span>            :               );</a>
<a name="100"><span class="lineNum">     100 </span>            :             } else {</a>
<a name="101"><span class="lineNum">     101 </span><span class="lineNoCov">          0 :               return CustomException(</span></a>
<a name="102"><span class="lineNum">     102 </span>            :                 exceptionType: _ExceptionType.Unauthorized,</a>
<a name="103"><span class="lineNum">     103 </span><span class="lineNoCov">          0 :                 statusCode: error.response?.statusCode,</span></a>
<a name="104"><span class="lineNum">     104 </span>            :                 message: 'Unauthorized user',</a>
<a name="105"><span class="lineNum">     105 </span><span class="lineNoCov">          0 :                 failureType: UnAuthorizedFailure(),</span></a>
<a name="106"><span class="lineNum">     106 </span>            :               );</a>
<a name="107"><span class="lineNum">     107 </span>            :             }</a>
<a name="108"><span class="lineNum">     108 </span><span class="lineNoCov">          0 :           case DioErrorType.other:</span></a>
<a name="109"><span class="lineNum">     109 </span><span class="lineNoCov">          0 :             if (error.message.contains(_ExceptionType.SocketException.name)) {</span></a>
<a name="110"><span class="lineNum">     110 </span><span class="lineNoCov">          0 :               return CustomException(</span></a>
<a name="111"><span class="lineNum">     111 </span>            :                 exceptionType: _ExceptionType.FetchDataException,</a>
<a name="112"><span class="lineNum">     112 </span><span class="lineNoCov">          0 :                 statusCode: error.response?.statusCode,</span></a>
<a name="113"><span class="lineNum">     113 </span>            :                 message: 'No internet connectivity',</a>
<a name="114"><span class="lineNum">     114 </span><span class="lineNoCov">          0 :                 failureType: NetworkFailure(),</span></a>
<a name="115"><span class="lineNum">     115 </span>            :               );</a>
<a name="116"><span class="lineNum">     116 </span>            :             }</a>
<a name="117"><span class="lineNum">     117 </span><span class="lineNoCov">          0 :             if (error.response?.statusCode == null) {</span></a>
<a name="118"><span class="lineNum">     118 </span><span class="lineNoCov">          0 :               return CustomException(</span></a>
<a name="119"><span class="lineNum">     119 </span>            :                 exceptionType: _ExceptionType.UnrecognizedException,</a>
<a name="120"><span class="lineNum">     120 </span><span class="lineNoCov">          0 :                 statusCode: error.response?.statusCode,</span></a>
<a name="121"><span class="lineNum">     121 </span><span class="lineNoCov">          0 :                 message: error.response?.statusMessage ?? 'Unknown',</span></a>
<a name="122"><span class="lineNum">     122 </span><span class="lineNoCov">          0 :                 failureType: UnRecognizedFailure(),</span></a>
<a name="123"><span class="lineNum">     123 </span>            :               );</a>
<a name="124"><span class="lineNum">     124 </span>            :             }</a>
<a name="125"><span class="lineNum">     125 </span><span class="lineNoCov">          0 :             final name = error.response?.data['meta']['code'] as int;</span></a>
<a name="126"><span class="lineNum">     126 </span><span class="lineNoCov">          0 :             final message = error.response?.data['meta']['message'] as String;</span></a>
<a name="127"><span class="lineNum">     127 </span><span class="lineNoCov">          0 :             if (message == _ExceptionType.TokenExpiredException.name) {</span></a>
<a name="128"><span class="lineNum">     128 </span><span class="lineNoCov">          0 :               return CustomException(</span></a>
<a name="129"><span class="lineNum">     129 </span>            :                 exceptionType: _ExceptionType.TokenExpiredException,</a>
<a name="130"><span class="lineNum">     130 </span><span class="lineNoCov">          0 :                 code: name.toString(),</span></a>
<a name="131"><span class="lineNum">     131 </span><span class="lineNoCov">          0 :                 statusCode: error.response?.statusCode,</span></a>
<a name="132"><span class="lineNum">     132 </span>            :                 message: message,</a>
<a name="133"><span class="lineNum">     133 </span><span class="lineNoCov">          0 :                 failureType: UnAuthorizedFailure(),</span></a>
<a name="134"><span class="lineNum">     134 </span>            :               );</a>
<a name="135"><span class="lineNum">     135 </span>            :             }</a>
<a name="136"><span class="lineNum">     136 </span><span class="lineNoCov">          0 :             if (message == 'Unauthorized.') {</span></a>
<a name="137"><span class="lineNum">     137 </span><span class="lineNoCov">          0 :               return CustomException(</span></a>
<a name="138"><span class="lineNum">     138 </span>            :                 exceptionType: _ExceptionType.Unauthorized,</a>
<a name="139"><span class="lineNum">     139 </span><span class="lineNoCov">          0 :                 statusCode: error.response?.statusCode,</span></a>
<a name="140"><span class="lineNum">     140 </span>            :                 message: 'Unauthorized user',</a>
<a name="141"><span class="lineNum">     141 </span><span class="lineNoCov">          0 :                 failureType: UnAuthorizedFailure(),</span></a>
<a name="142"><span class="lineNum">     142 </span>            :               );</a>
<a name="143"><span class="lineNum">     143 </span>            :             }</a>
<a name="144"><span class="lineNum">     144 </span><span class="lineNoCov">          0 :             return CustomException(</span></a>
<a name="145"><span class="lineNum">     145 </span>            :               message: message,</a>
<a name="146"><span class="lineNum">     146 </span><span class="lineNoCov">          0 :               code: name.toString(),</span></a>
<a name="147"><span class="lineNum">     147 </span><span class="lineNoCov">          0 :               statusCode: error.response?.statusCode,</span></a>
<a name="148"><span class="lineNum">     148 </span>            :             );</a>
<a name="149"><span class="lineNum">     149 </span>            :         }</a>
<a name="150"><span class="lineNum">     150 </span>            :       } else {</a>
<a name="151"><span class="lineNum">     151 </span><span class="lineNoCov">          0 :         return CustomException(</span></a>
<a name="152"><span class="lineNum">     152 </span>            :           exceptionType: _ExceptionType.UnrecognizedException,</a>
<a name="153"><span class="lineNum">     153 </span>            :           message: 'Error unrecognized',</a>
<a name="154"><span class="lineNum">     154 </span>            :         );</a>
<a name="155"><span class="lineNum">     155 </span>            :       }</a>
<a name="156"><span class="lineNum">     156 </span><span class="lineNoCov">          0 :     } on FormatException catch (e) {</span></a>
<a name="157"><span class="lineNum">     157 </span><span class="lineNoCov">          0 :       return CustomException(</span></a>
<a name="158"><span class="lineNum">     158 </span>            :         exceptionType: _ExceptionType.FormatException,</a>
<a name="159"><span class="lineNum">     159 </span><span class="lineNoCov">          0 :         message: e.message,</span></a>
<a name="160"><span class="lineNum">     160 </span><span class="lineNoCov">          0 :         failureType: FormatFailure(),</span></a>
<a name="161"><span class="lineNum">     161 </span>            :       );</a>
<a name="162"><span class="lineNum">     162 </span><span class="lineNoCov">          0 :     } on Exception catch (_) {</span></a>
<a name="163"><span class="lineNum">     163 </span><span class="lineNoCov">          0 :       return CustomException(</span></a>
<a name="164"><span class="lineNum">     164 </span>            :         exceptionType: _ExceptionType.UnrecognizedException,</a>
<a name="165"><span class="lineNum">     165 </span>            :         message: 'Error unrecognized',</a>
<a name="166"><span class="lineNum">     166 </span>            :       );</a>
<a name="167"><span class="lineNum">     167 </span>            :     }</a>
<a name="168"><span class="lineNum">     168 </span>            :   }</a>
<a name="169"><span class="lineNum">     169 </span>            : </a>
<a name="170"><span class="lineNum">     170 </span><span class="lineNoCov">          0 :   factory CustomException.fromParsingException(Exception error) {</span></a>
<a name="171"><span class="lineNum">     171 </span><span class="lineNoCov">          0 :     debugPrint('$error');</span></a>
<a name="172"><span class="lineNum">     172 </span><span class="lineNoCov">          0 :     return CustomException(</span></a>
<a name="173"><span class="lineNum">     173 </span>            :       exceptionType: _ExceptionType.SerializationException,</a>
<a name="174"><span class="lineNum">     174 </span>            :       message: 'Failed to parse network response to model or vice versa',</a>
<a name="175"><span class="lineNum">     175 </span>            :     );</a>
<a name="176"><span class="lineNum">     176 </span>            :   }</a>
<a name="177"><span class="lineNum">     177 </span>            : </a>
<a name="178"><span class="lineNum">     178 </span><span class="lineNoCov">          0 :   @override</span></a>
<a name="179"><span class="lineNum">     179 </span><span class="lineNoCov">          0 :   List&lt;Object&gt; get props =&gt; [exceptionType, message];</span></a>
<a name="180"><span class="lineNum">     180 </span>            : </a>
<a name="181"><span class="lineNum">     181 </span><span class="lineNoCov">          0 :   @override</span></a>
<a name="182"><span class="lineNum">     182 </span>            :   bool get stringify =&gt; true;</a>
<a name="183"><span class="lineNum">     183 </span>            : }</a>
</pre>
      </td>
    </tr>
  </table>
  <br>

  <table width="100%" border=0 cellspacing=0 cellpadding=0>
    <tr><td class="ruler"><img src="../../../../../glass.png" width=3 height=3 alt=""></td></tr>
    <tr><td class="versionInfo">Generated by: <a href="http://ltp.sourceforge.net/coverage/lcov.php" target="_parent">LCOV version 1.15.alpha0w</a></td></tr>
  </table>
  <br>

</body>
</html>
